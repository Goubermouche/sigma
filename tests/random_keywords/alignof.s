i32 main() {
	printf("%d\n", alignof(i8));
	printf("%d\n", alignof(i16));
	printf("%d\n", alignof(i32));
	printf("%d\n", alignof(i64));
	printf("%d\n", alignof(u8));
	printf("%d\n", alignof(u16));
	printf("%d\n", alignof(u32));
	printf("%d\n", alignof(u64));
	printf("%d\n", alignof(bool));
	printf("%d\n", alignof(char));
	printf("%d\n", alignof(i32*));
	printf("%d\n", alignof(i32**));

	struct s1 {
		i32 a;
	};

	printf("%d\n", alignof(s1));
	
	struct s2 {
		i32 a;
		char** b;
	};

	printf("%d\n", alignof(s2));
    
	ret 0;
}
