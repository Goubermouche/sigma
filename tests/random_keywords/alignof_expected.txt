1
2
4
8
1
2
4
8
1
1
8
8
4
8
